app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -55,8 +55,14 @@ class ObligationService(gformConnector: GformConnector) {
       obligDetails <- taxResponse.obligation.obligations
       info <- makeAllInfoList(
                taxResponse.id,
-               idNumbers.find(hTPWithId => hTPWithId.hmrcTaxPeriod == taxResponse.id).get,
-               obligDetails.obligationDetails)
+               idNumbers
+                 .find(hTPWithId => hTPWithId.hmrcTaxPeriod == taxResponse.id)
+                 .getOrElse(
```

Comment:
This doesn't make ObligationService safer. It makes it more obscure. We need to embrace and error (if there is no other way).

By embracing the error I mean that if we have method which we wish to return `List[TaxPeriodInformation]` type, but there is some element preventing this from being safe to do it, we need declare this method as partial in its return type.

One popular option for declaring method partial is `Either` data type. For example `Either[String, List[TaxPeriodInformation]]` is saying that method either give us back `List[TaxPeriodInformation]` or some `String` describing error condition.

Nice introductory materials are to be found in: https://fsharpforfunandprofit.com/rop/